ksqldb-common/src/main/java/io/confluent/ksql/util/KsqlConfig.java

```diff
@@ -949,4 +949,9 @@ public KsqlConfig overrideBreakingConfigsWithOriginalValues(final Map<String, ?>
     SslConfigs.addClientSslSupport(sslConfig);
     return sslConfig.names();
   }
+
+  public static String getServiceId(KsqlConfig ksqlConfig) {
+    return ReservedInternalTopics.KSQL_INTERNAL_TOPIC_PREFIX
+        + ksqlConfig.getString(KsqlConfig.KSQL_SERVICE_ID_CONFIG);
+  }
```

Comment:
This method is not returning what the name suggests.  It's not returning the service id, (which would be just `ksqlConfig.getString(KsqlConfig.KSQL_SERVICE_ID_CONFIG)`.

Instead, its returning the service id prefixed with `KSQL_INTERNAL_TOPIC_PREFIX`.

I see you've moved this from another class, but I still think this isn't what we want.  `KsqlConfig` doesn't need to know about `ReservedInternalTopics`.

This is only needed to construct the query application id. Would you mind moving code into a `QueryApplicationId` util class? e.g. 

```java
/**
 * Util for creating query application ids.
 */
public final class QueryApplicationId {

  private QueryApplicationId() {
  }

  public static String getQueryApplicationId(
      final KsqlConfig config,
      final boolean persistent,
      final QueryId queryId
  ) {
    final String serviceId = config.getString(KsqlConfig.KSQL_SERVICE_ID_CONFIG);

    final String configName = persistent
        ? KsqlConfig.KSQL_PERSISTENT_QUERY_NAME_PREFIX_CONFIG
        : KsqlConfig.KSQL_TRANSIENT_QUERY_NAME_PREFIX_CONFIG;
    
    final String queryPrefix = config.getString(configName);
    
    return ReservedInternalTopics.KSQL_INTERNAL_TOPIC_PREFIX
        + serviceId
        + queryPrefix
        + queryId;
  }
}
```

Thereby decoupling `KqlConfig` from any notion of internal topics, and also streamlining the existing code.